src/public/client.ts

```diff
@@ -41,32 +41,74 @@ class DisplayMap {
         <small class="form-text text-muted" style="display: inline-block;">PORT: ${data.port}</small>
         <hr />
         <div class="deleteButton" href="/">Delete</div>
-        <div class="deleteButton edit" href="/" style="padding: 0px 0px 0px 20px;">Edit</div>
+        <div class="edit" href="/" style="padding: 0px 0px 0px 20px;">Edit</div>
     </li>
 `
 
       const delButton = helper.getElement('.deleteButton', mappingElement)
       delButton.onclick = (): void => {
-        fetch(`/api/mappings/delete/${data.id}`, {
-          method: 'DELETE',
-          body: JSON.stringify({ data }),
-          headers: {
-            'Content-Type': 'application/json'
-          }
-        }).then(() => {
-          window.location.reload()
-        })
+        if (confirm('Are you sure want to delete this domain?')) {
+          fetch(`/api/mappings/delete/${data.id}`, {
+            method: 'DELETE',
+            body: JSON.stringify({ data }),
+            headers: {
+              'Content-Type': 'application/json'
+            }
+          }).then(() => {
+            window.location.reload()
+          })
+        }
       }
       const editButton = helper.getElement('.edit', mappingElement)
       //TODO: turn into input boxes for ability to edit.
       editButton.onclick = (): void => {
-        fetch(`/api/mappings/edit/${data.id}`, {
-          method: 'PATCH',
-          body: JSON.stringify({ data }),
-          headers: {
-            'Content-Type': 'application/json'
-          }
-        })
+        mappingElement.innerHTML = `
+        <li class='list-group-item'>
```

Comment:
add `style="display: flex;"`, refer to the some `li` element above to maintain consistency